### PR TITLE
update sdk to v 0.0.3:

### PR DESCRIPTION
**Pull request by interstellarPotato**
- add signOnly()
- add verifyOnly()
- update peacemakr-core-crypto to v 0.2.2

### CODE REVIEW

File: src/main/java/io/peacemakr/crypto/impl/crypto/ICryptoImpl.java [hunk 1]
```diff
@@ -696,10 +706,70 @@ private AsymmetricKey getSigningKey(SymmetricKeyUseDomain useDomain) {
...
+    Gson gson = new Gson();
+    byte[] addStringBytes = gson.toJson(aad).getBytes(StandardCharsets.UTF_8);
+
+    // Message digest
+    MessageDigest digest = MessageDigest.SHA_256;
+
+    byte[] signedBlob = Crypto.signAsymmetric(signingPrivateKey, plainText, addStringBytes, digest);
+    // Crypto.signAsymmetric might return null if failed to create a signing key
+    if (signedBlob == null)
+      throw new PeacemakrException("Failed to sign plain text");
```
Comment (notasecret) [inline src/main/java/io/peacemakr/crypto/impl/crypto/ICryptoImpl.java:752]: for the future, any way to help make errors more actionable?

File: src/main/java/io/peacemakr/crypto/impl/crypto/ICryptoImpl.java [hunk 2]
```diff
@@ -696,10 +706,70 @@ private AsymmetricKey getSigningKey(SymmetricKeyUseDomain useDomain) {
...
+      throw new PeacemakrException("Provided signed blob array object can not be empty");
+    }
+
+    // Private key
+    AsymmetricKey signingPrivateKey = getAsymmetricPrivateKey();
+
+    byte[] verifiedBlob = Crypto.verifyAsymmetric(signingPrivateKey, signedBlob);
+    // null is returned if failed to verify
+    if (verifiedBlob == null)
+      throw new PeacemakrException("Failed to verify signed blob");
```
Comment (notasecret) [inline src/main/java/io/peacemakr/crypto/impl/crypto/ICryptoImpl.java:771]: "failed to verify" vs "verification failed" might be important distinctions at some point?

File: src/main/java/io/peacemakr/crypto/impl/crypto/ICryptoImpl.java (hunk 1 above)
Comment (interstellarPotato) [reply src/main/java/io/peacemakr/crypto/impl/crypto/ICryptoImpl.java:752]: JNI will need to change, which is deff the way to go.

File: src/main/java/io/peacemakr/crypto/ICrypto.java [hunk 3]
```diff
@@ -62,7 +62,25 @@
   byte[] decrypt(byte[] cipherText) throws PeacemakrException;
 
   /**
-   * For visibility or debugging purposes, returns a string whihc identifies which
+   * Sing the message.
```
Comment (notasecret) [inline src/main/java/io/peacemakr/crypto/ICrypto.java:65]: *sign

Also, have to be super clear that the blob contains the originally signed message

File: src/main/java/io/peacemakr/crypto/ICrypto.java [hunk 4]
```diff
@@ -62,7 +62,25 @@
...
+   * <p>
+   * @param plainText clear text.
```
Comment (notasecret) [inline src/main/java/io/peacemakr/crypto/ICrypto.java:67]: *plaintext

File: src/main/java/io/peacemakr/crypto/ICrypto.java [hunk 5]
```diff
@@ -62,7 +62,25 @@
...
+   * @return A signedBlob upon success.
+   * @throws PeacemakrException if an exception occurs during signing attempt.
```
Comment (notasecret) [inline src/main/java/io/peacemakr/crypto/ICrypto.java:69]: what happens if verification fails?

File: src/main/java/io/peacemakr/crypto/ICrypto.java [hunk 6]
```diff
@@ -62,7 +62,25 @@
...
+   */
+  byte[] signOnly(byte[] plainText) throws PeacemakrException;
+
+  /**
+   * Verify the signed blob. VerifyOnly only works on blob returned by the result of SignOnly().
+   * <p>
+   * @param signedBlob packaged blob containing unencrypted data.
+   * @return message in the clear upon success.
```
Comment (notasecret) [inline src/main/java/io/peacemakr/crypto/ICrypto.java:77]: We mean, "... upon successful verification" maybe? ... and.... what happens when verification is not successful?

**Review by notasecret — APPROVED** [review]
nice, few minor questions

File: src/main/java/io/peacemakr/crypto/ICrypto.java (hunk 6 above)
Comment (interstellarPotato) [reply src/main/java/io/peacemakr/crypto/ICrypto.java]: This is an example of what c lib + JNI will produce if verify fails:
```
asymmetric_verify: 197 - error:00000000:lib(0):func(0):reason(0)
asymmetric_verify: 198 - DigestVerifyFinal failed
Verification of the message failed
``` 

the returned byte[] will be null, hence checking for null before returning. If C.verify() returns null, javaSDK throws an exception

**Review by yhung119 — APPROVED** [review]
LGTM :) many thanks

**Pull request merged**